packages/edge-login-ui-rn/src/components/themed/ThemedInputModal.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import * as React from 'react'
+import { type AirshipBridge } from 'react-native-airship'
+
+import s from '../../common/locales/strings.js'
+import { type ThemeProps, withTheme } from '../services/ThemeContext.js'
+import { EdgeTextField } from './EdgeTextField.js'
+import { ModalMessage, ModalTitle } from './ModalParts.js'
+import { PrimaryButton, SecondaryButton } from './ThemedButtons.js'
+import { ThemedModal } from './ThemedModal.js'
+
+type OwnProps = {
+  bridge: AirshipBridge<string | void>,
+  buttonLabel: string,
+  inputLabel: string,
+  message: string,
+  isInputPassword?: boolean,
+  onSubmit(inputText: string): Promise<string | void>,
+  title: string
+}
+
+type State = {
+  input: string,
+  error?: string
+}
+
+type Props = OwnProps & ThemeProps
+
+class ThemedInputModalComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      input: ''
+    }
+  }
+
+  handleInputChange = (input: string) => {
+    this.setState({ input })
+  }
+
+  handleSubmit = () => {
+    const { onSubmit, bridge } = this.props
+    this.setState({ error: undefined })
+    onSubmit(this.state.input)
+      .then((result: string | void) => {
+        if (result) {
```

Comment:
The logic should be `if (typeof result === 'string') { this.setState({error: result}) } else { bridge.resolve(this.state.input) }`